src/sparse/merkle_tree.rs

```diff
@@ -26,25 +26,25 @@ impl<StorageError> From<StorageError> for MerkleTreeError<StorageError> {
     }
 }
 
-pub struct MerkleTree<'storage, StorageType> {
+pub struct MerkleTree<StorageType> {
     root_node: Node,
-    storage: &'storage mut StorageType,
+    storage: StorageType,
```

Comment:
It is perfectly fine to just use generic here and not reference, as Storage implements `&mut` for its trait here: https://github.com/FuelLabs/fuel-storage/blob/21319d8412edda0dc1387ca201408ccb8c112d6b/src/lib.rs#L43
And rust will be able to match both `storage` and `&mut storage` bcs of that

Reply:
If I understand correctly, this allows us to specify that `storage` member is either an owned type or a non-owning reference type based on the constructor argument. We can construct the standard Merkle tree by providing a storage reference. We can define an in-memory Merkle tree by constructing a standard Merkle tree with an owned argument. Very cool!

Reply:
Without the explicit `'storage` lifetime, Rust simply knows that `'a` lives as long as or longer than the lifetime of `&mut storage`?

Reply:
I tried to find more concreat explanation but couldn't find any. In essence `StorageType` is owner in MerkleTree and its lifetime is tied to that structure, soo when giving mut ref as `StorageType`, rust compiler needs to check lifetime of `&mut storage` and created object `merkletree`. At least this is my understanding.